services/kvstore/src/main/java/tech/pegasys/pantheon/services/kvstore/RocksDBStats.java

```diff
@@ -0,0 +1,198 @@
+    TickerType.NUMBER_MULTIGET_CALLS,
+    TickerType.NUMBER_FILTERED_DELETES,
+    TickerType.NUMBER_MERGE_FAILURES,
+    TickerType.BLOOM_FILTER_PREFIX_CHECKED,
+    TickerType.BLOOM_FILTER_PREFIX_USEFUL,
+    TickerType.NUMBER_OF_RESEEKS_IN_ITERATION,
+    TickerType.GET_UPDATES_SINCE_CALLS,
+    TickerType.BLOCK_CACHE_COMPRESSED_MISS,
+    TickerType.BLOCK_CACHE_COMPRESSED_HIT,
+    TickerType.BLOCK_CACHE_COMPRESSED_ADD,
+    TickerType.BLOCK_CACHE_COMPRESSED_ADD_FAILURES,
+    TickerType.WAL_FILE_SYNCED,
+    TickerType.WAL_FILE_BYTES,
+    TickerType.WRITE_DONE_BY_SELF,
+    TickerType.WRITE_DONE_BY_OTHER,
+    TickerType.WRITE_TIMEDOUT,
+    TickerType.WRITE_WITH_WAL,
+    TickerType.COMPACT_READ_BYTES,
+    TickerType.COMPACT_WRITE_BYTES,
+    TickerType.FLUSH_WRITE_BYTES,
+    TickerType.NUMBER_DIRECT_LOAD_TABLE_PROPERTIES,
+    TickerType.NUMBER_SUPERVERSION_ACQUIRES,
+    TickerType.NUMBER_SUPERVERSION_RELEASES,
+    TickerType.NUMBER_SUPERVERSION_CLEANUPS,
+    TickerType.NUMBER_BLOCK_COMPRESSED,
+    TickerType.NUMBER_BLOCK_DECOMPRESSED,
+    TickerType.NUMBER_BLOCK_NOT_COMPRESSED,
+    TickerType.MERGE_OPERATION_TOTAL_TIME,
+    TickerType.FILTER_OPERATION_TOTAL_TIME,
+    TickerType.ROW_CACHE_HIT,
+    TickerType.ROW_CACHE_MISS,
+    TickerType.READ_AMP_ESTIMATE_USEFUL_BYTES,
+    TickerType.READ_AMP_TOTAL_READ_BYTES,
+    TickerType.NUMBER_RATE_LIMITER_DRAINS,
+    TickerType.NUMBER_ITER_SKIP,
+    TickerType.NUMBER_MULTIGET_KEYS_FOUND,
+  };
+
+  // Histograms - treated as prometheus summaries
+  static final HistogramType[] HISTOGRAMS = {
+    HistogramType.DB_GET,
+    HistogramType.DB_WRITE,
+    HistogramType.COMPACTION_TIME,
+    HistogramType.SUBCOMPACTION_SETUP_TIME,
+    HistogramType.TABLE_SYNC_MICROS,
+    HistogramType.COMPACTION_OUTFILE_SYNC_MICROS,
+    HistogramType.WAL_FILE_SYNC_MICROS,
+    HistogramType.MANIFEST_FILE_SYNC_MICROS,
+    HistogramType.TABLE_OPEN_IO_MICROS,
+    HistogramType.DB_MULTIGET,
+    HistogramType.READ_BLOCK_COMPACTION_MICROS,
+    HistogramType.READ_BLOCK_GET_MICROS,
+    HistogramType.WRITE_RAW_BLOCK_MICROS,
+    HistogramType.STALL_L0_SLOWDOWN_COUNT,
+    HistogramType.STALL_MEMTABLE_COMPACTION_COUNT,
+    HistogramType.STALL_L0_NUM_FILES_COUNT,
+    HistogramType.HARD_RATE_LIMIT_DELAY_COUNT,
+    HistogramType.SOFT_RATE_LIMIT_DELAY_COUNT,
+    HistogramType.NUM_FILES_IN_SINGLE_COMPACTION,
+    HistogramType.DB_SEEK,
+    HistogramType.WRITE_STALL,
+    HistogramType.SST_READ_MICROS,
+    HistogramType.NUM_SUBCOMPACTIONS_SCHEDULED,
+    HistogramType.BYTES_PER_READ,
+    HistogramType.BYTES_PER_WRITE,
+    HistogramType.BYTES_PER_MULTIGET,
+    HistogramType.BYTES_COMPRESSED,
+    HistogramType.BYTES_DECOMPRESSED,
+    HistogramType.COMPRESSION_TIMES_NANOS,
+    HistogramType.DECOMPRESSION_TIMES_NANOS,
+    HistogramType.READ_NUM_MERGE_OPERANDS,
+  };
+
+  static void registerRocksDBMetrics(
+      final Statistics stats, final PrometheusMetricsSystem metricsSystem) {
```

Comment:
If you move this to the `metrics` package and inject `MetricsCategory`, we can reuse this elsewhere.  For example, in the `RocksDbTaskQueue`.

Reply:
That would give the metrics package a dependency on RocksDB, and then every package that uses metrics would drag that dependency along as deadweight.

Reply:
Yeah, thats a good point.  And that's the case now with vertx.  I guess the "right" thing to do would be to have independent `vertxmetrics` and `rocksdbmetrics` packages.